src/borg/archive.py

```diff
@@ -1142,7 +1164,7 @@ def item_is_hardlink_master(item):
                     'source' not in item and
                     not matcher.match(item.path))
 
-        for item in archive.iter_items():
+        for item in archive.iter_items(filter=lambda item: 'checkpoint' not in item):
```

Comment:
I thought about that, but we might add support for listing / extracting these checkpoint archives (e.g. via a cmdline option).